### ISSUE TITLE
'finishedBuilder', 'resetBuilder' don't fire

### ISSUE THREAD

**Issue by Apollo108** [federicodesia/custom_timer#2]
Hi, thank you for the awesome Timer, it's really customizable, but I have one issue.
According to the code, these builders will fire only when controller is provided. Why?
![image](https://user-images.githubusercontent.com/16933202/103912984-adac2a00-5110-11eb-80ec-a03d5e47be54.png)
![image](https://user-images.githubusercontent.com/16933202/103913232-0380d200-5111-11eb-8cfd-57c3bc77070d.png)

**Comment by federicodesia** [comment]
Hi,  I'm glad that is helpful to you!😃

I forgot to add it in the documentation, but even though a controller is not provided, this widget internally uses a CustomTimerController for its operation.

You can find it in the widget controller definition:
```
/// Controls the state of the timer. Ensure you dispose of it when done.
///
/// If null, this widget will create its own [CustomTimerController] and dispose it when the widget is disposed.
final CustomTimerController controller;
```

**Comment by Apollo108** [comment]
Fine, but this doesn't solve the problem. However, as soon as I add an external controller, everything works fine.
So maybe there is any reason it works that way, and do you plan to change this behaviour soon?

**Comment by federicodesia** [comment]
So I don't understand the problem. Please detail it so I will try to help you.

**Comment by Apollo108** [comment]
The problem is that according to your code (see the screenshot), 'finishedBuilder' will never fire until we provide widget.controller, i.e. - external controller. As long as widget.controller is null, widget.controller?.state == CustomTImerState.finished condition will never be true, and callback won't fire.

**Comment by federicodesia** [comment]
You're right! I had not realized this. I fixed it in version 0.0.3.
Thank you very much for helping to improve this widget! 🚀

**Comment by federicodesia** [comment]
You're right! I had not realized this. I fixed it in version 0.0.3.
Thank you very much for helping to improve this widget! 🚀

**Comment by Apollo108** [comment]
> You're right! I had not realized this. I fixed it in version 0.0.3.
> Thank you very much for helping to improve this widget! 🚀

Thank you again for the awesome customizable timer, good luck with the professional life)
